src/main/java/com/quizapp/enterprise/services/QuizService.java

```diff
@@ -34,19 +37,24 @@ public Quiz getQuiz(int quizId) throws Exception {
 
         Optional<Quiz> quizToReturn =  quizRepository.findById(quizId);
         if(!quizToReturn.isPresent()){
-            throw new Exception("Quiz with id "  + quizId +  "does not exist");
-
+            StringBuilder missingQuizException = new StringBuilder("Quiz with id ")
+                    .append(quizId)
+                    .append("does not exist");
+            throw new Exception(missingQuizException.toString());
         }else{
             return quizToReturn.get();
         }
 
     }
 
     @Override
-    public ArrayList<Quiz> getAllQuizzes(String username){
-        int userId =  userRepository.findUserByEmail(username).getUserID();
+    public ArrayList<Quiz> getAllQuizzes(String username) throws Exception {
+        User user = userRepository.findUserByEmail(username);
+        if(user == null){
+            throw new Exception("user not found");
+        }
```

Comment:
included a check for a possible null prior to execution on possible null object so code may handle the case as desired.
This change was recommended by the "8. Proper handling of Null Pointer Exceptions"  section of  the java coding best practices summary referenced below

source:https://xperti.io/blogs/java-coding-best-practices/